File: Data Structures & Algorithms/duplicate-integer/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        const d = {};
        for (const i of nums) {
            if (d[i]) return true;
            d[i] = 1;
        }
        return false;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-0.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        vector<bool> visit(nums.size() + 1, false);
        for (int n : nums) {
            if (n > 0 && n <= nums.size()) visit[n] = true;
        }
        for (int i = 1; i < nums.size() + 1; i++) {
            if (!visit[i]) return i;
        }
        return nums.size() + 1;
    }
};